src/esm_parser/yaml_to_dict.py

```diff
@@ -49,6 +55,106 @@ def __init__(self, fpath, yaml_error):
         super().__init__(self.message)
 
 
+def str_to_bool(string):
+    """Handy function to cast strings to bools. Eg.
+    bool('false') -> True
+    str_to_bool('false') -> False
+    str_to_bool('0') -> False
+    """
+    truthy_values = ["true", "1"]
+    return string.lower() in truthy_values
+
+
+def node_to_value(node):
+    """Converts a PyYAML node to its corresponding value
+
+    Parameters
+    ----------
+    node : yaml.Node
+
+    Returns
+    -------
+    Any supported Python scalar data type (eg. int, float, bool)
+
+    Raises
+    ------
+    TypeError
+        - When input is not a PyYAML node
+        - When type to cast is not supported
+    """
+    if not isinstance(node, yaml.Node):
+        raise TypeError(f"{node} ({type(node).__qualname__}) is not a YAML Node object")
+
+    # eg. 'tag:yaml.org,2002:int'
+    tag = node.tag.split(":")[-1]
+    tag_to_constructor_map = {
+        "int": int,
+        "float": float,
+        "str": str,
+        "bool": str_to_bool,
+    }
+    constructor = tag_to_constructor_map.get(tag, None)
+    if constructor is None:
+        raise TypeError(f"{tag} can not be used as a constructor")
+    return constructor(node.value)
```

Comment:
Is this related to the problem this PR is trying to solve? If not I'd suggest to move it to a separate PR.